src/Middleware/RedirectHandler.cs

```diff
@@ -86,7 +86,7 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                         }
 
                         // general clone request with internal CloneAsync (see CloneAsync for details) extension method
-                        var newRequest = await response.RequestMessage.CloneAsync();
+                        var newRequest = await response.RequestMessage!.CloneAsync();

```

Comment:
we could add a condition with a type assertion instead of a bang operator. It'd be safer for scenarios like mocking etc...